src/test/regress/sql/full_join.sql

```diff
@@ -0,0 +1,91 @@
+--
+-- Full join with subquery pushdown support 
+--
+
+SET citus.next_shard_id TO 9000000;
+
+CREATE TABLE test_table_1(id int, val1 int);
+CREATE TABLE test_table_2(id bigint, val1 int);
+CREATE TABLE test_table_3(id int, val1 bigint);
+
+SELECT create_distributed_table('test_table_1', 'id');
+SELECT create_distributed_table('test_table_2', 'id');
+SELECT create_distributed_table('test_table_3', 'id');
+
+INSERT INTO test_table_1 VALUES(1,1),(2,2),(3,3);
+INSERT INTO test_table_2 VALUES(2,2),(3,3),(4,4);
+INSERT INTO test_table_3 VALUES(1,1),(3,3),(4,5);
+
+-- Simple full outer join
+SELECT id FROM test_table_1 FULL JOIN test_table_3 using(id) ORDER BY 1;
+
+-- Get all columns as the result of the full join
+SELECT * FROM test_table_1 FULL JOIN test_table_3 using(id) ORDER BY 1;
+
+-- Join subqueries using single column
+SELECT * FROM 
+	(SELECT test_table_1.id FROM test_table_1 FULL JOIN test_table_3 using(id)) as j1
+		FULL JOIN
+	(SELECT test_table_1.id FROM test_table_1 FULL JOIN test_table_3 using(id)) as j2
+	USING(id)
+	ORDER BY 1;
+
+-- Join subqueries using multiple columns
+SELECT * FROM 
+	(SELECT test_table_1.id, test_table_1.val1 FROM test_table_1 FULL JOIN test_table_3 using(id)) as j1
+		FULL JOIN
+	(SELECT test_table_1.id, test_table_1.val1 FROM test_table_1 FULL JOIN test_table_3 using(id)) as j2
+	USING(id, val1)
+	ORDER BY 1;
+
+-- Full join using multiple columns
+SELECT * FROM test_table_1 FULL JOIN test_table_3 USING(id, val1) ORDER BY 1;
```

Comment:
Maybe change one of the tests to have more complicated target list:
```SQL
SELECT count(DISTINCT id), avg(test_table_1.val1) + id * id, id::numeric IS NOT NULL  FROM test_table_1 FULL JOIN test_table_3 using(id) WHERE id::bigint < 55 GROUP BY id ORDER BY 2 ASC LIMIT 3;
```

Or, anything such.